src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/BookerPrisonerInfoClient.kt

```diff
@@ -28,39 +26,40 @@ class BookerPrisonerInfoClient(
   }
 
   fun getPermittedPrisonerInfo(
-    offenderSearchPrisoner: PrisonerDto,
     bookerPrisoner: PermittedPrisonerForBookerDto,
   ): BookerPrisonerInfoDto? {
-    val prisonCode = offenderSearchPrisoner.prisonId!!
-    val prisonMono = visitSchedulerClient.getPrisonAsMono(prisonCode)
-    val visitBalancesDtoMono = prisonApiClient.getVisitBalancesAsMono(offenderSearchPrisoner.prisonerNumber)
-
-    Mono.zip(prisonMono, visitBalancesDtoMono).block(apiTimeout).also {
-      val prison = it?.t1?.getOrNull()
+    val prisonCode = bookerPrisoner.prisonCode
+    val prisonerId = bookerPrisoner.prisonerId
+    val offenderSearchPrisonerDtoMono = prisonerSearchClient.getPrisonerByIdAsMonoEmptyIfNotFound(prisonerId)
+    val visitBalancesDtoMono = prisonApiClient.getVisitBalancesAsMono(prisonerId)
+    val registeredPrisonMono = prisonRegisterClient.getPrisonAsMonoEmptyIfNotFound(prisonCode)
+    Mono.zip(offenderSearchPrisonerDtoMono, visitBalancesDtoMono, registeredPrisonMono).block(apiTimeout).also {
+      val offenderSearchPrisoner = it?.t1
       val visitBalancesDto = it?.t2?.getOrNull()
-      if (prison != null) {
-        publicBookerValidationUtil.validatePrison(prison)?.let { msg ->
-          LOG.error(
-            "getPermittedPrisonerInfo - ${
-              MessageFormat.format(
-                PRISON_VALIDATION_ERROR_MSG,
-                prisonCode,
-                bookerPrisoner.prisonerId,
-                msg,
-              )
-            }",
-          )
-        } ?: return BookerPrisonerInfoDto(
+      val registeredPrison = getRegisteredPrison(prisonCode, it?.t3?.getOrNull())
+
+      return if (offenderSearchPrisoner == null) {
+        LOG.error("getPermittedPrisonerInfo - prisoner with id - $prisonerId not found on offender search")
+        null
+      } else {
+        return BookerPrisonerInfoDto(
           offenderSearchPrisoner,
           visitBalancesUtil.calculateAvailableVos(visitBalancesDto),
           visitBalancesUtil.calculateVoRenewalDate(visitBalancesDto),
+          registeredPrison,
         )
-      } else {
-        LOG.error("getPermittedPrisonerInfo Prison with code - $prisonCode, not enabled on visit-scheduler")
-        return null
       }
     }
+  }
 
-    return null
+  private fun getRegisteredPrison(prisonCode: String, prisonDto: PrisonRegisterPrisonDto?): RegisteredPrisonDto {
+    return if (prisonDto != null) {
+      RegisteredPrisonDto(prisonDto)
+    } else {
+      RegisteredPrisonDto(
+        prisonCode = prisonCode,
+        prisonName = prisonCode,
```

Comment:
prisonCode = prisonCode,
prisonName = prisonCode,

Both being set to prisonCode, is this expected?

Reply:
this is to ensure that prisonName is not being returned as null to the front end on the rare occasion where prison details are returned as null from prison register.